### PR TITLE
STM32U0:  provide support for DAC peripherals

### PR DESCRIPTION
**Pull request by djiatsaf-st**
This PR adds digital to analog converter (DAC) peripherals  on the stm32U0  boards.

### CODE REVIEW

File: tests/drivers/dac/dac_api/src/test_dac.c [hunk 1]
```diff
@@ -4,107 +4,80 @@
...
+	defined(CONFIG_BOARD_STM32F3_DISCO) || defined(CONFIG_BOARD_NUCLEO_F429ZI) ||              \
+	defined(CONFIG_BOARD_NUCLEO_F746ZG) || defined(CONFIG_BOARD_NUCLEO_F767ZI) ||              \
+	defined(CONFIG_BOARD_NUCLEO_F722ZE) || defined(CONFIG_BOARD_NUCLEO_G071RB) ||              \
+	defined(CONFIG_BOARD_NUCLEO_G431RB) || defined(CONFIG_BOARD_NUCLEO_G474RE) ||              \
+	defined(CONFIG_BOARD_NUCLEO_H743ZI) || defined(CONFIG_BOARD_NUCLEO_L073RZ) ||              \
+	defined(CONFIG_BOARD_NUCLEO_L152RE) || defined(CONFIG_BOARD_DISCO_L475_IOT1) ||            \
+	defined(CONFIG_BOARD_NUCLEO_L552ZE_Q) || defined(CONFIG_BOARD_STM32L562E_DK) ||            \
+	defined(CONFIG_BOARD_STM32H573I_DK) || defined(CONFIG_BOARD_B_U585I_IOT02A) ||             \
+	defined(CONFIG_BOARD_NUCLEO_U575ZI_Q) || defined(CONFIG_BOARD_NUCLEO_U5A5ZJ_Q) ||          \
+	defined(CONFIG_BOARD_NUCLEO_WL55JC) || defined(CONFIG_BOARD_NUCLEO_U083RC) ||              \
```
Comment (erwango) [inline tests/drivers/dac/dac_api/src/test_dac.c:21]: Clang-format is not always right

Please re-instantiate the previous format which was easier to read in this specific case.
For instance it was easy to see which board is added and exactly where it should be added.
With this format, adding one board will require to change all lines

Apply everywhere

File: boards/st/nucleo_u031r8/nucleo_u031r8.dts [hunk 2]
```diff
@@ -81,3 +81,9 @@
 	ahb-prescaler = <1>;
 	apb1-prescaler = <1>;
 };
+
+&dac1{
```
Comment (erwango) [inline boards/st/nucleo_u031r8/nucleo_u031r8.dts:85]: Please update board documentation ("supported features")

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by martinjaeger — COMMENTED** [review]
Looks good, only got minor comments.

Can you please squash the last commit into the second one, so that the docs are updated together with the board `.dts` file?

**Review by martinjaeger — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

File: boards/st/nucleo_u031r8/doc/index.rst
Comment (erwango) [inline boards/st/nucleo_u031r8/doc/index.rst:1]: Please update board's .yaml file

**Review by erwango — CHANGES_REQUESTED** [review]
Last catch otherwise LGTM, thanks

**Review by martinjaeger — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by martinjaeger — DISMISSED** [review]

**Comment by erwango** [comment]
@djiatsaf-st To fix the issue of tests overflowing soc footprint, don't hesitate to add `    min_ram: xx` to impacted test cases yaml files. This will be helpful for others as well.

**Review by gautierg-st — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by marwaiehm-st — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

File: boards/st/nucleo_u031r8/nucleo_u031r8.dts [hunk 3]
```diff
@@ -82,6 +82,12 @@
 	apb1-prescaler = <1>;
 };
 
+&dac1{
```
Comment (mathieuchopstm) [inline boards/st/nucleo_u031r8/nucleo_u031r8.dts:85]: Nit:
```suggestion
&dac1 {
```

File: boards/st/nucleo_u083rc/nucleo_u083rc.dts [hunk 4]
```diff
@@ -82,6 +82,13 @@
 	apb1-prescaler = <1>;
 };
 
+
+&dac1{
```
Comment (mathieuchopstm) [inline boards/st/nucleo_u083rc/nucleo_u083rc.dts:86]: Nit:
```suggestion
&dac1 {
```

File: boards/st/stm32u083c_dk/stm32u083c_dk.dts [hunk 5]
```diff
@@ -72,6 +72,12 @@
 	apb1-prescaler = <1>;
 };
 
+&dac1{
```
Comment (mathieuchopstm) [inline boards/st/stm32u083c_dk/stm32u083c_dk.dts:75]: Nit:
```suggestion
&dac1 {
```

**Review by mathieuchopstm — DISMISSED** [review]
LGTM. Minor non-blocking details:

File: boards/st/nucleo_u083rc/nucleo_u083rc.dts [hunk 6]
```diff
@@ -82,6 +82,13 @@
 	apb1-prescaler = <1>;
 };
 
+
```
Comment (martinjaeger) [inline boards/st/nucleo_u083rc/nucleo_u083rc.dts:85]: Unnecessary empty line.

**Review by martinjaeger — CHANGES_REQUESTED** [review]
Now that @mathieuchopstm identified the missing spaces, let's get all minor style issues fixed for consistency. Afterwards it's definitely ready.

**Review by martinjaeger — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**